R/osmactive.R

```diff
@@ -781,6 +781,62 @@ get_bus_routes = function(
 }
 
 
+#' Function to assign default speed limits
+#' 
+#' 
+#' @param osm_data An sf object with the road network
+#' @return An sf object with the default speed limits in the column `maxspeed`
+#' @export
+#' 
+get_maxspeed_uk = function(osm_data) {
+  # Define default speed limits (in mph) based on highway types
+  default_speed_limits = c(
+    motorway = "70 mph",
+    motorway_link = "70 mph",
+    trunk = "60 mph",
+    trunk_link = "60 mph",
+    primary = "60 mph",
+    primary_link = "60 mph",
+    secondary = "60 mph",
+    secondary_link = "60 mph",
+    tertiary = "60 mph",
+    tertiary_link = "60 mph",
+    unclassified = "60 mph",
+    residential = "30 mph",
+    living_street = "20 mph",
+    service = "30 mph",
+    track = "15 mph",
+    pedestrian = "5 mph",
+    footway = "5 mph",
+    path = "5 mph",
+    cycleway = "15 mph",
+    steps = "5 mph",
+    construction = "5 mph",
+    road = "30 mph"
+  )
```

Comment:
If you make `default_speeds_uk()` a function it would be more modular and these lines could be replaced by:

```
get_maxspeed_uk = function(osm_data, default_speeds = default_speeds_uk()) {